app/models/job_invocation_composer.rb

```diff
@@ -0,0 +1,217 @@
+class JobInvocationComposer
+  attr_accessor :params, :job_invocation, :host_ids, :search_query
+  attr_reader :job_template_ids
+
+  # TODO aka initialize_from_params
+  def initialize(job_invocation, params)
```

Comment:
I was thinking about creating `app/services` or something similar but it seems overcomplicating for one file... Since I don't think it should be in helpers (usually contains modules) or controllers (these should only take inputs and return outputs from other objects) so remaining option is models which usually holds the logic. Since this object cooperates with other models I think models/ is OK. Also you can view having same params as in controller being lucky coincidence, when we receive parameters from API, API controller should convert those to composer accepted format first. Does it make sense?